youtube_dl/extractor/cbc.py

```diff
@@ -249,13 +275,19 @@ def _valid_device_token(self):
         return self._device_id and self._device_token
 
     def _register_device(self):
+        signature = self._signature()
         self._device_id = self._device_token = None
         result = self._download_xml(
             self._API_BASE_URL + 'device/register',
             None, 'Acquiring device token',
             data=b'<device><type>web</type></device>')
         self._device_id = xpath_text(result, 'deviceId', fatal=True)
         self._device_token = xpath_text(result, 'deviceToken', fatal=True)
+        if signature:
+            data = '<login><token>{0}</token><device><deviceId>{1}</deviceId><type>web</type></device></login>'.format(signature, self._device_id).encode()
```

Comment:
Are you sure `signature`, `self._device_id` will not have `<`, `>`? Strictly speaking those should be escaped or better standard XML facilities should be used for XML building.